packages/playground/src/components/select_farm.vue

```diff
@@ -79,15 +80,20 @@ const SIZE = 20;
 const page = ref();
 const farmInput = useInputRef();
 const profileManager = useProfileManager();
-const country = ref<string>();
+const location = ref(createLocation());
 const search = ref<string>();
 const searchInput = ref<string>();
 const farm = ref<FarmInterface>();
 const farmManager = useFarm();
-watch([farm, country], ([f, c]) => {
-  farmManager?.setFarmId(f?.farmID);
-  emits("update:modelValue", f ? { farmID: f.farmID, name: f.name, country: c ?? undefined } : undefined);
-});
+watch(
+  [farm, location],
+  ([f, c]) => {
+    const farm = { farmID: f?.farmID || -1, name: f?.name || defaultFarm, country: c.country, region: c.region };
```

Comment:
Well defaulting to undefined here requires converting `FarmInterface` to a partial `farmId?: number | undefined` however it might be used somewhere else but incase of -1 that `select-farm` component won't be used outside select-node so it's fine to handle `-1` rather than `undefined`